### PR TITLE
Add new SolveBio "Dashboard" to variant detail view

### PR DESCRIPTION
**Pull request by davecap**
Remove SolveBio requests from variant resource.
Bundles in the SolveBio JS library which loads the dashboard iframe.
Using the new dashboard view requires the solvebio python module >= 1.5.2 and app credentials.
The JS view obtains a SolveBio OAuth2 token from a view within the django_solvebio app (part of the solvebio module). The Dashboard runs in an iframe which is a locally rendered template (variant-detail.html).
This code is still very rough around the edges and updates will be made in the next few days.

Signed-off-by: David Caplan dcaplan@gmail.com

### CODE REVIEW

**Comment by naegelyd** [comment]
@davecap Is the iFrame that is being loaded remote(different domain) or the same domain as Varify itself is running on? I've just started reviewing this so I apologize if this is made very obvious by some portion of the changes included in this PR.

**Comment by davecap** [comment]
No problem! Thanks for checking out the PR. Right now, there's a view in `django_solvebio` (part of the `solvebio` Python package) that basically just serves Django templates for iframes. The code for that is not actually in the PR, but is in the latest version of our Python package.

File: varify/conf/global_settings.py [hunk 1]
```diff
@@ -405,12 +404,14 @@
     INSTALLED_APPS += ('solvebio.contrib.django_solvebio', )
 
 
-# Get your API key from https://www.solvebio.com/account
-SOLVEBIO_API_KEY = os.environ.get('SOLVEBIO_API_KEY', None)
+# SolveBio OAuth2 Application credentials
+SOLVEBIO_APP_ID = os.environ.get('SOLVEBIO_APP_ID', None)
```
Comment (bruth) [inline varify/conf/global_settings.py]: Calling `get` defaults to `None`, so using `None` as the default value is redundant.

Comment (davecap) [inline varify/conf/global_settings.py]: fixed in the latest commit

**Comment by davecap** [comment]
I just pushed a change that loads the iframe from an alternate domain. For now it's a public S3 domain, but we will work on a private URL that requires a valid access token.

File: varify/static/js/src/ui/result-details.js [hunk 2]
```diff
@@ -131,6 +186,23 @@ define([
             // properties always having values of 0 making it impossible to
             // detect overflow anywhere but here.
             this._checkForOverflow();
+
+            // Set up the SolveBio dashboard
+            var dashboardAttrs = this.model.attributes;
+            if (SolveBio.getAccessToken()) {
+                createSolveBioDashboard(dashboardAttrs);
+            } else if (!SolveBio.disabled) {
```
Comment (naegelyd) [inline varify/static/js/src/ui/result-details.js]: Please move `else if (!SolveBio.disabled) {` to next line

File: varify/static/js/src/ui/result-details.js [hunk 3]
```diff
@@ -131,6 +186,23 @@ define([
...
+                // Get SolveBio access token
+                $.getJSON('/api/solvebio/oauth2/token', function(data) {
+                    if (data.access_token) {  // jshint ignore:line
+                        SolveBio.setAccessToken(data.access_token);  // jshint ignore:line
+                        createSolveBioDashboard(dashboardAttrs);
+                    } else {
```
Comment (naegelyd) [inline varify/static/js/src/ui/result-details.js]: Just a formatting tweak like the comment above but please move the `else {` to the next line

**Comment by naegelyd** [comment]
@davecap Can you rebase this against master as some of these affected files have been changed since you opened this PR? Thanks.

**Comment by davecap** [comment]
It seems a bit messy, would you prefer if I created a new pull request?

**Comment by naegelyd** [comment]
Sure. That's fine.

**Comment by davecap** [comment]
Replaced with https://github.com/cbmi/varify/pull/384

**Pull request closed without merging**